cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -254,23 +255,31 @@ Given(/'(.+)' registers for the '(.+)' event using the '(.+)' API$/, async (appI
             assert(false, CONSTANTS.NO_MATCHED_RESPONSE);
           }
           result = JSON.parse(result);
-          cy.log(`Response from ${appId}: ${JSON.stringify(result.report.eventListenerResponse)}`);
-
-          // If event and params are not supported setting isScenarioExempted as true for further validation.
-          if (UTILS.isScenarioExempted(event, param)) {
-            Cypress.env(CONSTANTS.IS_SCENARIO_EXEMPTED, true);
+          cy.log(`Response from ${appId}: ${JSON.stringify(result.result)}`);
+          if (result && result.result && result.result.hasOwnProperty(CONSTANTS.LISTENING)) {
+            const eventResponse = {
+              eventListenerId: result.result.event + '-' + result.id,
+              eventListenerResponse: result.result,
+            };
+            result.result = eventResponse;
           }
```

Comment:
what is the purpose of this change? I believe all this if else conditions are handled in updateResponseForFCS already ?

Reply:
why do we need an if condition directly inside glue code

Reply:
As discussed in call